### DISCUSSION TITLE
[Request] Prometheus

### DISCUSSION THREAD

**Discussion by DSDV** [tteck/Proxmox#310] [Ideas]
Hi,

id like to request a new script for Prometheus.
https://prometheus.io/
It is a good database and monitoring tool but its main feature is the pull approach.
https://prometheus.io/docs/introduction/first_steps/

big thanks <3

**Comment by tteck** [comment]
Would you test
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/prometheus.sh)"
```
Please report your findings

**Comment by DSDV** [comment]
![image](https://user-images.githubusercontent.com/5504857/176970418-5e0b5308-f9fa-4c0b-ac53-7643096cac46.png)

sorry for the delay.  and thank you for the fast response. 
i tested the install and need to get some instant of blocky to run so i can better test the setup :D 

but from what i can tell it seems to run as expected (at the moment only tested with power monitoring <3
if i encounter any problems ill report back

**Comment by DSDV** [comment]
what i noticed (but not exclusive to this script) that you turn of the firewall for the network adapters. 
is there a reason you do this by default?

**Reply by tteck** [reply]
It's not that I turn it off, I just don't enable it. I leave that up to the user

**Reply by DSDV** [reply]
hm .. strange when i add network adapters (via gui) they are always firewalled by default o.o 
guess its a different default when using cli ?

**Reply by tteck** [reply]
When scripting, there's no defaults, it's what you add

**Reply by DSDV** [reply]
oh ok i see (: 
than it was my mistake expecting the gui defaults ^^

**Reply by tteck** [reply]
Yeah, "expecting" kills projects

**Reply by DSDV** [reply]
until now i didn't encounter any problems regarding your script :D 
thanks a lot for the implementation

is there any way to donate (for example to push the blocky implementation?) <3

**Reply by tteck** [reply]
This https://google.github.io/blockly/?

**Reply by tteck** [reply]
found it https://github.com/tteck/Proxmox/discussions/309

**Reply by DSDV** [reply]
yes exactly :D

**Comment by Masked-Kunsiquat** [comment]
Sorry to necro an old post (not fully comfortable with github yet).

Where would I find the `prometheus.yml` file in the newly created LXC? 

I created a new file and verified it with `promtool`; not sure how to point the installation to the new file for use. Or preferably, I can edit the default `prometheus.yml` if I find the location.


I've used your scripts to get a jump-start on my homelab & it's been amazing! This is probably easy to figure out, but I'm struggling. I really appreciate all your work!! (:

**Reply by Masked-Kunsiquat** [reply]
2 more google searches and I found my answer - sorry! Thanks again for the great scripts!!

In-case anyone needs it in the future --> `/etc/prometheus/prometheus.yml`

**Comment by towerhand** [comment]
Hey @tteck. Are there any plans to add an update path for this application? Thanks.

**Reply by tteck** [reply]
You can test
```
RELEASE=$(curl -s https://api.github.com/repos/prometheus/prometheus/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
wget -q https://github.com/prometheus/prometheus/releases/download/v${RELEASE}/prometheus-${RELEASE}.linux-amd64.tar.gz
tar -xf prometheus-${RELEASE}.linux-amd64.tar.gz
cd prometheus-${RELEASE}.linux-amd64
mv -f prometheus promtool /usr/local/bin/
mv -f consoles/ console_libraries/ /etc/prometheus/
```
Make sure you have a backup!

**Reply by towerhand** [reply]
mv: cannot move 'consoles/' to '/etc/prometheus/consoles': Directory not empty
mv: cannot move 'console_libraries/' to '/etc/prometheus/console_libraries': Directory not empty

**Reply by tteck** [reply]
Oops change `mv` to `cp`, sorry.

**Reply by towerhand** [reply]
cp: -r not specified; omitting directory 'consoles/'
cp: -r not specified; omitting directory 'console_libraries/'

It seems to work with cp -rf.

**Reply by tteck** [reply]
It was `mv -f` should have been `cp -f`

**Reply by towerhand** [reply]
```RELEASE=$(curl -s https://api.github.com/repos/prometheus/prometheus/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
wget -q https://github.com/prometheus/prometheus/releases/download/v${RELEASE}/prometheus-${RELEASE}.linux-amd64.tar.gz
tar -xf prometheus-${RELEASE}.linux-amd64.tar.gz
cd prometheus-${RELEASE}.linux-amd64
cp -rf prometheus promtool /usr/local/bin/
cp -rf consoles/ console_libraries/ /etc/prometheus/
cd
rm -r prometheus-${RELEASE}.linux-amd64
rm prometheus-${RELEASE}.linux-amd64.tar.gz

**Reply by tteck** [reply]
That works? Should I add that as an update path?

**Reply by tteck** [reply]
```
RELEASE=$(curl -s https://api.github.com/repos/prometheus/prometheus/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
wget -q https://github.com/prometheus/prometheus/releases/download/v${RELEASE}/prometheus-${RELEASE}.linux-amd64.tar.gz
tar -xf prometheus-${RELEASE}.linux-amd64.tar.gz
cd prometheus-${RELEASE}.linux-amd64
cp -rf prometheus promtool /usr/local/bin/
cp -rf consoles/ console_libraries/ /etc/prometheus/
cd ~
rm -rf prometheus-${RELEASE}.linux-amd64 prometheus-${RELEASE}.linux-amd64.tar.gz
```

**Reply by towerhand** [reply]
It seems to work on my end with no errors.

**Reply by tteck** [reply]
Done. Thank you for your help!

**Reply by towerhand** [reply]
Thank you. I appreciate all your scripts.
